webapp/src/components/github_repo_selector/github_repo_selector.jsx

```diff
@@ -30,25 +33,65 @@ export default class GithubRepoSelector extends PureComponent {
     }
 
     componentDidMount() {
-        this.props.actions.getRepos();
+        this.props.actions.getOrgs();
+        this.props.actions.getReposByOrg("");
     }
 
-    onChange = (_, name) => {
-        const repo = this.props.yourRepos.find((r) => r.full_name === name);
+    onChangeForOrg = (_, org) => {
+        if (this.state.org != org) {
+            this.setState({org: org}, () => {this.props.actions.getReposByOrg(org)});
+            this.props.onChange(null);
+        }
+    }
+
+    onChangeForRepo = (_, name) => {
+        const repo = this.props.yourReposByOrg.find((r) => r.full_name === name);
         this.props.onChange({name, permissions: repo.permissions});
     }
 
     render() {
-        const repoOptions = this.props.yourRepos.map((item) => ({value: item.full_name, label: item.full_name}));
+        const orgOptions  = this.props.yourOrgs.map((item) => ({value: item.login, label: item.login}));
+        orgOptions.unshift({value: "", label: "your owned repositories"});
```

Comment:
Maybe instead just have the user's login as the first option in the "org" selector, and have it automatically selected

Reply:
Yes, it will be automatically selected because the option's value is an empty string, and our org state is set to empty by default. The 'Users Login' option should be the first option because, for instance, if a user selects 'org1' initially and later wants to select 'Your Owned Repositories,' they should have the flexibility to choose that option.

please let me know if I am missing something.

Reply:
@mickmister any comment on this ?

Reply:
I was thinking that we would replace the label of this option `your owned repositories`, with the user's login instead. Same general functionality as current, but show the label as e.g `thoratvinod`

Reply:
Done.